apis/python/src/tiledb/vector_search/object_readers/directory_reader.py

```diff
@@ -35,30 +262,33 @@ def id(self) -> int:
 class DirectoryReader:
     def __init__(
         self,
-        uri: str,
-        glob: str = "**/[!.]*",
-        exclude: Sequence[str] = (),
-        suffixes: Optional[Sequence[str]] = None,
+        search_uri: str,
         config: Optional[Mapping[str, Any]] = None,
+        include: str = "*",
+        exclude: Sequence[str] = ["[.]*", "*/[.]*"],
+        suffixes: Optional[Sequence[str]] = None,
+        max_files: Optional[int] = None,
```

Comment:
I believe we need to accept a context or config here in the class constructor for use in subsequent TileDB calls.

Reply:
See comment for removing context